src/format.rs

```diff
@@ -165,15 +161,24 @@ impl Formatter5424 {
     }
 }
 
-impl<T: Display> LogFormat<(u32, StructuredData, T)> for Formatter5424 {
+impl<T: Display> LogFormat<(Option<String>, StructuredData, T)> for Formatter5424 {
     fn format<W: Write>(
         &self,
         w: &mut W,
         severity: Severity,
-        log_message: (u32, StructuredData, T),
+        log_message: (Option<String>, StructuredData, T),
     ) -> Result<()> {
         let (message_id, data, message) = log_message;
 
+        // XXX: seems a lot of effort per-call, we could do this via a wrapper type instead
+        // So the caller could do this once and pass it in
+        let message_id = message_id
+            .unwrap_or_else(|| NILL_VALUE.to_owned())
+            .chars()
+            .filter(is_us_print_ascii)
+            .take(32)
```

Comment:
We could also validate more of these fields to prevent the user from giving us values that violate the rfc, but at a runtime cost (as I've implemented here) https://github.com/ctrlaltf24/rust-syslog/commit/0362d743200a9127364013e1ada73ba01975c695#diff-c875d3cb38acba1839738726b3d4aa920bec536459b52d5bc5069a91dd1620f5R187-R218 . Not sure if you'd want those changes, assuming no since the additional runtime complexity